File: src/brs/grpc/handlers/GetTransactionBytesHandler.java
package brs.grpc.handlers;

import brs.Blockchain;
import brs.TransactionProcessor;
import brs.grpc.GrpcApiHandler;
import brs.grpc.proto.BrsApi;
import com.google.protobuf.ByteString;

public class GetTransactionBytesHandler implements GrpcApiHandler<BrsApi.GetTransactionRequest, BrsApi.TransactionBytes> {

    private final Blockchain blockchain;
    private final TransactionProcessor transactionProcessor;

    public GetTransactionBytesHandler(Blockchain blockchain, TransactionProcessor transactionProcessor) {
        this.blockchain = blockchain;
        this.transactionProcessor = transactionProcessor;
    }

    @Override
    public BrsApi.TransactionBytes handleRequest(BrsApi.GetTransactionRequest request) throws Exception {
        return BrsApi.TransactionBytes.newBuilder()
                .setTransactionBytes(ByteString.copyFrom(GetTransactionHandler.getTransaction(blockchain, transactionProcessor, request).getBytes()))
                .build();
    }
}

File: src/main/java/brs/grpc/handlers/GetTransactionBytesHandler.kt
package brs.grpc.handlers

import brs.DependencyProvider
import brs.grpc.GrpcApiHandler
import brs.grpc.proto.BrsApi
import brs.grpc.proto.ProtoBuilder
import brs.grpc.proto.toByteString

class GetTransactionBytesHandler(private val dp: DependencyProvider) : GrpcApiHandler<BrsApi.BasicTransaction, BrsApi.TransactionBytes> {

    override fun handleRequest(request: BrsApi.BasicTransaction): BrsApi.TransactionBytes {
        return BrsApi.TransactionBytes.newBuilder()
                .setTransactionBytes(ProtoBuilder.parseBasicTransaction(dp, request).bytes.toByteString())
                .build()
    }
}
